src/main/scala/org/ergoplatform/local/ErgoMiner.scala

```diff
@@ -225,51 +236,55 @@ class ErgoMiner(ergoSettings: ErgoSettings,
 
 object ErgoMiner extends ScorexLogging {
 
-  def createCoinbase(state: UtxoStateReader,
-                     feeBoxes: Seq[ErgoBox],
-                     minerProp: Value[SBoolean.type],
-                     emissionRules: EmissionRules): ErgoTransaction = {
+  /**
+    * Generate from 0 to 2 transaction, that collect rewards from fee boxes in block transactions `txs` and
+    * emission box from `state`
+    */
+  def collectRewards(state: UtxoStateReader,
+                     txs: Seq[ErgoTransaction],
+                     minerPk: ProveDlog,
+                     emissionRules: EmissionRules): Seq[ErgoTransaction] = {
     val emissionBoxOpt = state.emissionBoxOpt
     emissionBoxOpt foreach { emissionBox =>
       assert(state.boxById(emissionBox.id).isDefined, s"Emission box ${Algos.encode(emissionBox.id)} missed")
     }
-    val h = state.stateContext.lastHeaderOpt.map(_.height).getOrElse(-1)
-    createCoinbase(emissionBoxOpt, h, feeBoxes, minerProp, emissionRules)
+    collectRewards(emissionBoxOpt, state.stateContext.currentHeight, txs, minerPk, emissionRules)
   }
 
-  def createCoinbase(emissionBoxOpt: Option[ErgoBox],
+  /**
+    * Generate from 0 to 2 transaction, that collect rewards from fee boxes in block transactions `txs` and
+    * emission box `emissionBoxOpt`
+    */
+  def collectRewards(emissionBoxOpt: Option[ErgoBox],
                      currentHeight: Int,
-                     feeBoxes: Seq[ErgoBox],
-                     minerProp: Value[SBoolean.type],
-                     emission: EmissionRules): ErgoTransaction = {
-    feeBoxes.foreach(b => assert(b.proposition == TrueLeaf, s"Trying to create coinbase from protected fee box $b"))
+                     txs: Seq[ErgoTransaction],
+                     minerPk: ProveDlog,
+                     emission: EmissionRules): Seq[ErgoTransaction] = {
+    val feeBoxes: Seq[ErgoBox] = ErgoState.boxChanges(txs)._2.filter(_.proposition == Constants.FeeProposition)
```

Comment:
Better to compare _.propositionBytes (the comparison would be faster)

Reply:
Is it always faster? If yes - I propose to redefine `equals` in `sigmastate.Values.Value`.